src/main/java/seedu/address/logic/commands/DeleteLessonCommand.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CLASS_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LESSON_INDEX;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.components.name.Name;
+import seedu.address.model.lesson.Lesson;
+import seedu.address.model.moduleclass.ModuleClass;
+
+/**
+ * Deletes a lesson identified using it's displayed index in the displayed module class list.
+ */
+public class DeleteLessonCommand extends Command {
+
+    public static final String COMMAND_WORD = "delete-lesson";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the lesson in a specific module class identified by the "
+            + "index number used in the displayed module class list and lesson list respectively.\n"
```

Comment:
Perhaps it will be better to use `class` since we are using `class` in user guide?
```suggestion
            + ": Deletes the lesson in a specific class identified by the "
            + "index number used in the displayed class list and lesson list respectively.\n"
```